internal/k8s/converter.go

```diff
@@ -112,16 +99,12 @@ func NewUnstructuredConverter() (*UnstructuredConverter, error) {
 	return uc, nil
 }
 
-func (c *UnstructuredConverter) CanConvert(obj interface{}) bool {
-	_, ok := obj.(*unstructured.Unstructured)
-	return ok
-}
-
-// Convert converts an unstructured.Unstructured to typed struct
+// Convert converts an unstructured.Unstructured to typed struct. If obj
+// is not an unstructured.Unstructured it is returned without further processing.
 func (c *UnstructuredConverter) Convert(obj interface{}) (interface{}, error) {
 	unstructured, ok := obj.(*unstructured.Unstructured)
 	if !ok {
-		return nil, fmt.Errorf("unsupported object type: %T", obj)
+		return obj, nil
```

Comment:
If `Convert` passes through objects that it doesn't handle, should it also do so in the `default` switch case?

Reply:
I don't think so. If the object isn't an unstructured, we return it. If it _is_ an unstructured we try to decode it, and if that fails we balk.